app/services/dttp/access_token_service.rb

```diff
@@ -0,0 +1,44 @@
+module Dttp
+  class AccessTokenService
+    class Client
```

Comment:
I got a bit confused by this `Client` which seems to be responsible for authenticating and the other `Client` which is responsible for creating. I wonder whether it would make sense to make this a little clearer either in the class name or some other approach.

Reply:
Ah right... that other benefit 🤣

Reply:
I've renamed.